File: readability/readability.c
#include <ctype.h>
#include <cs50.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

float lws[] = {0, 0, 0};

void countLWS(string text);

int main(void)
{
    string text = get_string("Text: ");
    countLWS(text);

    float index = 0.0588 * ((lws[0]/lws[1]) * 100) - 0.296 * ((lws[2]/lws[1]) * 100) - 15.8;

    int grade = (int) round(index);

    if(grade < 0)
    {
        printf("Before Grade 1\n");
    } else if(grade > 16)
    {
        printf("Grade 16+\n");
    } else
    {
        printf("Grade %i\n", grade);
    }
}

void countLWS(string text)
{
    for(int i = 0, n = strlen(text); i < n; i++)
    {
        if(tolower(text[i]) >= 'a' && tolower(text[i]) <= 'z')
        {
            lws[0]++;
        } else if(text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            lws[2]++;
        } else if(text[i] == ' ')
        {
            lws[1]++;
        }
    }
    lws[1]++;
}

File: sentimental-readability/readability.py
lws = [0, 0, 0]


def main():
    text = input("Text: ")
    countLWS(text)

    index = 0.0588 * ((lws[0]/lws[1]) * 100) - 0.296 * ((lws[2]/lws[1]) * 100) - 15.8

    grade = int(round(index))

    if grade < 0:
        print("Before Grade 1")
    elif grade > 16:
        print("Grade 16+")
    else:
        print(f"Grade {grade}")


def countLWS(text):
    for i in text:
        if i.lower() >= 'a' and i.lower() <= 'z':
            lws[0] += 1
        elif i == '.' or i == '!' or i == '?':
            lws[2] += 1
        elif i == ' ':
            lws[1] += 1
    lws[1] += 1


main()
